core/arch/arm/plat-stm32mp1/rng_seed.c

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2020, Linaro Limited
+ */
+
+#include <assert.h>
+#include <drivers/stm32_rng.h>
+#include <drivers/stm32mp1_rcc.h>
+#include <io.h>
+#include <kernel/delay.h>
+#include <kernel/panic.h>
+#include <mm/core_memprot.h>
+#include <platform_config.h>
+#include <stm32_util.h>
+#include <tee/tee_cryp_utl.h>
+#include <trace.h>
+
+#define RNG1_RESET_TIMEOUT_US		1000
+#define PRNG_SEED_SIZE			16
+
+/* Override weak plat_rng_init with platform handler to seed PRNG */
+void plat_rng_init(void)
+{
+	vaddr_t rng = (vaddr_t)phys_to_virt(RNG1_BASE, MEM_AREA_IO_SEC);
+	vaddr_t rcc = stm32_rcc_base();
+	uint64_t timeout_ref = timeout_init_us(RNG1_RESET_TIMEOUT_US);
+	uint8_t seed[PRNG_SEED_SIZE] = { };
+	size_t size = 0;
+
+	assert(cpu_mmu_enabled());
+
+	/* Setup RNG1 without clock/reset driver support, not yet initialized */
+	io_setbits32(rcc + RCC_MP_AHB5ENSETR, RCC_MP_AHB5ENSETR_RNG1EN);
+	io_setbits32(rcc + RCC_MP_AHB5LPENCLRR, RCC_MP_AHB5LPENSETR_RNG1LPEN);
+	io_setbits32(rcc + RCC_AHB5RSTSETR, RCC_AHB5RSTSETR_RNG1RST);
+	while (!(io_read32(rcc + RCC_AHB5RSTSETR) & RCC_AHB5RSTSETR_RNG1RST))
+		if (timeout_elapsed(timeout_ref))
+			panic();
```

Comment:
nit: You can add proper error messages to the panic calls, making debugging in case of RNG failure easier.

Reply:
We really don't expect to panic here. The debug trace should be enough get file/line info.